src/engine/sidechain/enginebroadcast.cpp

```diff
@@ -12,53 +11,38 @@
 #undef WIN32
 #endif
 
-#include "engine/sidechain/enginebroadcast.h"
-
 #include "broadcast/defs_broadcast.h"
 #include "control/controlpushbutton.h"
-#include "encoder/encoder.h"
-#include "encoder/encoderbroadcastsettings.h"
-#include "mixer/playerinfo.h"
 #include "preferences/usersettings.h"
-#include "recording/defs_recording.h"
-
-#include "track/track.h"
-
-static const int kConnectRetries = 30;
-static const int kMaxNetworkCache = 491520;  // 10 s mp3 @ 192 kbit/s
-// Shoutcast default receive buffer 1048576 and autodumpsourcetime 30 s
-// http://wiki.shoutcast.com/wiki/SHOUTcast_DNAS_Server_2
-static const int kMaxShoutFailures = 3;
-
-EngineBroadcast::EngineBroadcast(UserSettingsPointer pConfig)
-        : m_pTextCodec(nullptr),
-          m_pMetaData(),
-          m_pShout(nullptr),
-          m_pShoutMetaData(nullptr),
-          m_iMetaDataLife(0),
-          m_iShoutStatus(0),
-          m_iShoutFailures(0),
-          m_settings(pConfig),
+#include "util/sample.h"
+
+#include "engine/sidechain/enginebroadcast.h"
+
+namespace {
+const int kNetworkLatencyFrames = 8192; // 185 ms @ 44100 Hz
+// Related chunk sizes:
+// Mp3 frames = 1152 samples
+// Ogg frames = 64 to 8192 samples.
+// In Mixxx 1.11 we transmit every decoder-frames at once,
+// Which results in case of ogg in a dynamic latency from 0.14 ms to to 185 ms
+// Now we have switched to a fixed latency of 8192 frames (stereo samples) =
+// which is 185 @ 44100 ms and twice the maximum of the max mixxx audio buffer
+const int kBufferFrames = kNetworkLatencyFrames * 4; // 743 ms @ 44100 Hz
+// normally * 2 is sufficient.
+// We allow to buffer two extra chunks for a CPU overload case, when
+// the broadcast thread is not scheduled in time.
+}
+
+EngineBroadcast::EngineBroadcast(UserSettingsPointer pConfig,
+                                 BroadcastSettingsPointer pBroadcastSettings,
+                                 QSharedPointer<EngineNetworkStream> pNetworkStream)
+        : m_settings(pBroadcastSettings),
           m_pConfig(pConfig),
-          m_encoder(nullptr),
-          m_pMasterSamplerate(new ControlProxy("[Master]", "samplerate")),
-          m_custom_metadata(false),
-          m_firstCall(false),
-          m_format_is_mp3(false),
-          m_format_is_ov(false),
-          m_protocol_is_icecast1(false),
-          m_protocol_is_icecast2(false),
-          m_protocol_is_shoutcast(false),
-          m_ogg_dynamic_update(false),
+          m_pNetworkStream(pNetworkStream),
```

Comment:
It looks like we can remove the network stream dependency here and just pass the channel count.